src/serial/serial.c

```diff
@@ -59,27 +63,52 @@ struct Serial {
 	struct serial_cfg cfg;
 };
 
+void serial_purge_rx_queue(struct Serial* s)
+{
+	xQueueReset(s->rx_queue);
+}
+
+void serial_purge_tx_queue(struct Serial* s)
+{
+	xQueueReset(s->tx_queue);
+}
+
+/**
+ * Clears the contents of the rx and tx queues.
+ */
+void serial_clear(struct Serial *s)
+{
+        serial_purge_rx_queue(s);
+        serial_purge_tx_queue(s);
+}
+
+/**
+ * Unblocks our Seral rx queue by putting an invalid character on the
+ * front of the queue. This will unblock any task that is waiting on
+ * the queue.  Its up to the rx handler below to ensure that we return
+ * the correct status code.
+ */
+static void unblock_rx_queue(struct Serial *s)
+{
+	xQueueSendToFront(s->rx_queue, &invalid_char, 0);
```

Comment:
Not quite.  I do this to unblock any tasks that are blocked by this queue.  The only way they can read data from this queue is the get_c method below.  In that method I check if the serial device is closed and if it saw the invalid character.  If that happens, then I return an error status indicating that the Serial device is closed instead of returning this invalid character.  I then also re-call this very method there to unblock any other tasks that may be blocked.  This ensures that no task remains blocked (no matter how many are waiting).